nservicebus/ravendb/unit-of-work.md

```diff
@@ -14,31 +14,55 @@ Share the session between the message handler(s) and the actual unit of work imp
 
 Instead, beginning with NServiceBus version 3, use the new [support for child containers](/nservicebus/containers/child-containers.md). This means that all dependencies configured as a single call effectively become static within the context of one transport message.
 
-To resolve a RavenDB document session from the container, add the following configuration (StructureMap is used but any of the other containers except Spring and Unity would work):
-
+To resolve a RavenDB document session from the container, add the following configuration (StructureMap is used but any of the other containers except Spring and Unity would work):  
 
 ```C#
 var store = new DocumentStore {Url = "http://localhost:8080", DefaultDatabase = "MyDatabase"};
 store.Initialize();
  
-ObjectFactory.Configure(c =>
-    {
-        c.For<IDocumentStore>()
-            .Singleton()
-            .Use(store);
-        c.For<IDocumentSession>()
-            .Use(ctx => ctx.GetInstance<IDocumentStore>()
-                        .OpenSession());
-        c.For<IManageUnitsOfWork>()
-            .Use<RavenUnitOfWork>();
-    });
- 
-Configure.With()
-            .StructureMapBuilder(ObjectFactory.Container);
+//ObjectFactory.Configure(c =>  // objectFactory has been deprecated in structureMap
+Container container = new Container( c => 
+                    {
+                        c.For<IDocumentStore>().Singleton()
+                            .Use(dbStore);
+                        c.For<IDocumentSession>()
+                            .Use(ctx => ctx.GetInstance<IDocumentStore>()
+                                        .OpenSession());
+                        c.For<IManageUnitsOfWork>()
+                            .Use<RavenUnitOfWork>();
+                    });
+configuration.UseContainer<StructureMapBuilder>(c =>
+                       c.ExistingContainer(container));
+
+// old version: 
```

Comment:
You won't believe how much time I wasted finding out that I can use the Container container = new Container...

There are tons of documentation asking about the obsolete ObjectFactory and all giving obscure and complex answers (by which you have to create your own ObjectFactory...  or by which you "shouldn't be using anything like it, that's why it went obsolete...)

When moving back to the non Unit of Work (because of the licensing issue) I forgot to remove the constructor in the handler, and my handler stopped working. After 3 hours of panic and two minutes after posting to NServiceBs support someone at NServiceBus pinpointed the problem. Yay! (and whew!)